### ISSUE TITLE
Everywhere in tide::Error that takes a StatusCode should take a TryInto<StatusCode>

### ISSUE THREAD

**Issue by jbr** [http-rs/tide#649]
## Detailed Description
Elsewhere in tide, interfaces that take a StatusCode also accept a numeric representation of that StatusCode. tide::Error interfaces do not yet.

Potentially, this could be broadened to a project-wide search for StatusCode and replacing it with `impl TryInto<StatusCode>` and `.try_into().unwrap()`.  Or possibly using this :point_down: and `impl Into<StatusCode>`

```rust
impl From<u16> for StatusCode {
    fn into(self: u16) -> StatusCode {
        self.try_into().unwrap()
    }
}
```

**Comment by yusuke-ota** [comment]
I think this is better to impl http-types::Error::new() like [http-types::Response::new](https://github.com/http-rs/http-types/blob/bbd63d8f5787d0eec38b6218098feefe34451a4b/src/response.rs#L94-L101).
May I make pull request to http-types for http-types::Error::new() and http-types::Error::from_str() enable to use from u16?

```rust
fn exsample(){
    let http_types_error = Error::new(404, io::Error::new(io::ErrorKind::Other, "Not Found"));
    assert_eq!(http_types_error.status(), StatusCode::NotFound);
}
```

**Comment by yusuke-ota** [comment]
Should not I make `http-types::Error::new` accept `u16`?

If there are better ways, please tell me.
I want to challenge this issue.

**Comment by jbr** [comment]
I'm not certain I understand your proposal. Is it to use TryFrom<StatusCode> for Error::new and Error::from_str? If so, sounds good!

**Comment by yusuke-ota** [comment]
Yes, I wanted to explain that "use TryInto<StatusCode> for http-types::Error::new and http-types::Error::from_str".
Sorry to my bad explanation.

**Comment by jbr** [comment]
Closing this because it was fixed in http-rs/http-types#216 and will be available in tide whenever the next http-types release is made. Thanks!
